source/AI.cpp

```diff
@@ -311,6 +352,72 @@ AI::AI(const PlayerInfo &player, const List<Ship> &ships,
 
 
 // Fleet commands from the player.
+void AI::IssueFormationChange(const PlayerInfo &player)
+{
+	// Figure out what ships we are giving orders to
+	vector<Ship *> targetShips = GetShipsForFormationCommand(player);
+	if(targetShips.empty())
+		return;
+
+	const auto &formationPatterns = GameData::Formations();
+	if(formationPatterns.empty())
+	{
+		Messages::Add("No formations available.", Messages::Importance::High);
```

Comment:
Since we're adding some vanilla formations, a console error would fit here better.

Reply:
I'm not sure; the player is issueing a command to change a formation in-game, so it seems sensible to also provide the feedback in-game (if there are no formations available, which is likely rare since we are indeed already adding some by default).